### PR TITLE
`Development`:  Increase client test coverage for contributor model

### PR DESCRIPTION
**Pull request by canberkanar**
### Checklist

- [X] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [X] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).

### Description

Added a client test to improve the test coverage of contributor-model.ts.

#### Code Review
- [x] Review 1
- [x] Review 2

### Test Coverage

 contributor-model.ts 100% | 0%  --> 100% | 100%

### CODE REVIEW

File: src/test/javascript/spec/component/about-us/contributor-model.spec.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+describe('ContributorModel', () => {
+    it('constructor should set class variables accordingly', () => {
+        const fullName = 'Full Name';
+        const photoDirectory = 'Photo Directory';
+        const role = 'ADMIN';
+        const website = 'www.website.de';
+
+        const model = new ContributorModel(fullName, photoDirectory, role, website);
+
+        expect(model.fullName).toEqual(fullName);
```
Comment (coaxial1) [inline src/test/javascript/spec/component/about-us/contributor-model.spec.ts:12]: We can use ``toBe`` in all 4 cases here

File: src/test/javascript/spec/component/about-us/contributor-model.spec.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { ContributorModel } from '../../../../../main/webapp/app/core/about-us/models/contributor-model';
```
[the remark this answers is not in the document]
Comment (canberkanar) [reply src/test/javascript/spec/component/about-us/contributor-model.spec.ts]: thx

File: src/test/javascript/spec/component/about-us/contributor-model.spec.ts (hunk 1 above)
Comment (canberkanar) [reply src/test/javascript/spec/component/about-us/contributor-model.spec.ts]: updated accordingly

**Comment by canberkanar** [comment]
> I left 2 comments in the code.
> 
> But to be honest: Do we really want to keep this test? This is not testing any functionality or business logic.
> 
> Instead, to add coverage to this model, I'd suggest to extend `about-us.component.spec.ts` to actually render one or two contributors into HTML using this model, and then assert that it's actually on the page, e.g. by looking for their full name.

Thank you for your suggestion. I will look into it.

**Comment by akesfeden** [comment]
I agree with @coaxial1 about extending `about-us.component.spec.ts` for this model. An alternative might be converting the `AboutUsModel` and `ContributorModel` classes to interfaces if they are not instantiated at all, but since they have -Model suffix I am not sure this approach would be better.

File: src/test/javascript/spec/component/about-us/contributor-model.spec.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (canberkanar) [reply src/test/javascript/spec/component/about-us/contributor-model.spec.ts]: @coaxial1 I updated the code according to your suggestions.

File: src/test/javascript/spec/component/about-us/about-us.component.spec.ts [hunk 3]
```diff
@@ -47,4 +48,29 @@ describe('AboutUsComponent', () => {
...
+        const fullName = 'Full Name';
+        const photoDirectory = 'Photo Directory';
+        const role = 'ADMIN';
+        const website = 'www.website.de';
+
+        const contributors = [new ContributorModel(fullName, photoDirectory, role, website)];
+
+        const getStaticJsonFromArtemisServerStub = jest.spyOn(staticContentService, 'getStaticJsonFromArtemisServer').mockReturnValue(of(new AboutUsModel([], contributors)));
+        const getProfileInfoSub = jest.spyOn(profileService, 'getProfileInfo');
+        getProfileInfoSub.mockReturnValue(
```
Comment (coaxial1) [inline src/test/javascript/spec/component/about-us/about-us.component.spec.ts:65]: I suggest to assert that this stub has been called in your ``then`` block below, like you did with the other stub.
Additionally, we can inline this (and change ``Sub`` to ``Stub``;D):
```suggestion
        const getProfileInfoStub = jest.spyOn(profileService, 'getProfileInfo').mockReturnValue(
```

File: src/test/javascript/spec/component/about-us/about-us.component.spec.ts [hunk 4]
```diff
@@ -47,4 +48,29 @@ describe('AboutUsComponent', () => {
...
+            new BehaviorSubject<ProfileInfo>({ inProduction: false, sshCloneURLTemplate: 'ssh://git@testserver.com:1234/' } as ProfileInfo).asObservable(),
```
Comment (coaxial1) [inline src/test/javascript/spec/component/about-us/about-us.component.spec.ts:66]:
```suggestion
            of({ inProduction: false, sshCloneURLTemplate: 'ssh://git@testserver.com:1234/' } as ProfileInfo)
```
You just want an observable, right? Or is there a reson to use a subject?

**Review by coaxial1 — CHANGES_REQUESTED** [review]

File: src/test/javascript/spec/component/about-us/about-us.component.spec.ts (hunk 3 above)
Comment (canberkanar) [reply src/test/javascript/spec/component/about-us/about-us.component.spec.ts]: Unfortunately, prettier does not permit me to inline as you demonstrated on line 64, however, I still got rid of using the variable to call spyOn() anyways. Also, sub is now stub :)

image

File: src/test/javascript/spec/component/about-us/about-us.component.spec.ts (hunk 4 above)
Comment (canberkanar) [reply src/test/javascript/spec/component/about-us/about-us.component.spec.ts]: Yes exactly. Nice trick 👍🏻

File: src/test/javascript/spec/component/about-us/about-us.component.spec.ts [hunk 5]
```diff
@@ -47,4 +48,29 @@ describe('AboutUsComponent', () => {
...
+        const getStaticJsonFromArtemisServerStub = jest.spyOn(staticContentService, 'getStaticJsonFromArtemisServer').mockReturnValue(of(new AboutUsModel([], contributors)));
+        const getProfileInfoStub = jest
+            .spyOn(profileService, 'getProfileInfo')
+            .mockReturnValue(of({ inProduction: false, sshCloneURLTemplate: 'ssh://git@testserver.com:1234/' } as ProfileInfo));
+
+        fixture.detectChanges();
+        tick();
+        fixture.whenStable().then(() => {
+            expect(getStaticJsonFromArtemisServerStub).toHaveBeenCalledTimes(1);
+            expect(getProfileInfoStub).toHaveBeenCalled();
```
[the remark this answers is not in the document]
Comment (canberkanar) [reply src/test/javascript/spec/component/about-us/about-us.component.spec.ts]: Addressed it, thanks. 👍🏻

**Comment by canberkanar** [comment]
> I agree with @coaxial1 about extending `about-us.component.spec.ts` for this model. An alternative might be converting the `AboutUsModel` and `ContributorModel` classes to interfaces if they are not instantiated at all, but since they have -Model suffix I am not sure this approach would be better.

@akesfeden I updated the code according to the suggestions. Can you take another look?

**Review by bassner — APPROVED** [review]
Looks very good to me now! Thanks for adding the changes. 👍

**Review by willergeorg — APPROVED** [review]
Test code lgtm!

**Review by filip-gregurevic — APPROVED** [review]
Looks good to me 👍🏼

**Pull request merged**